src/esbmc/main.cpp

```diff
@@ -3,9 +3,12 @@
 #include <langapi/mode.h>
 
 #include <irep2/irep2.h>
-
+#include <util/config.h>
 int main(int argc, const char **argv)
 {
   esbmc_parseoptionst parseoptions(argc, argv);
+  for (int i = 0; i < argc; i++)
+    config.args.push_back(argv[i]);
```

Comment:
why this is needed?

Reply:
You probably just need to get them from the cmdline object

Reply:
How would I access the contents of argv from the cmdline object? I have been trying to figure out how to do this recently but from what I have found the 'args' member of cmdline only contains the files passed and cmdline_options contains all options available to the ESBMC. I have also had a look at the options object, but the 'option_map' member also appears to use all the available ESBMC options. Is any of this correct?

Reply:
If it's not there, let's start by checking where we fill the options object.

Reply:
I believe that the options_map field is filled in in the cmdline method of the options object, which is called in esbmc_parseoptions. This uses the vm member of cmdline, which I believe is filled in using all_cmdline_options.

Reply:
If there is an easy way to access the contents of argv in the correct order then I can use that with some modifications of some important method signatures (if that is acceptable). Otherwise, would Rafael's code in question be an acceptable solution?

Reply:
I have now made some changes: argv and argc will now both be passed to html.cpp and the parameter information will now not be duplicated in the main file.